### PR TITLE
[Notification] Add notification cli server command

### PR DESCRIPTION
**Pull request by Sxnan**
## What is the purpose of the change

Add notification cli server command #170 

## Brief change log

- Add notification cli server command

## Verifying this change

This change added tests.

### CODE REVIEW

File: lib/notification_service/bin/start-notification.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ fi
 
 echo "Starting notification server"
 LOG_FILE_NAME=notification_server-$(date "+%Y%m%d-%H%M%S").log
-"$BIN"/start_notification_server.py > "${NOTIFICATION_LOG_DIR}"/"${LOG_FILE_NAME}" 2>&1 &
+notification server > "${NOTIFICATION_LOG_DIR}"/"${LOG_FILE_NAME}" 2>&1 &
 echo $! > "${NOTIFICATION_PID_FILE}"
```
Comment (wuchaochen) [inline lib/notification_service/bin/start-notification.sh:34]: notification server start/stop/init

File: lib/notification_service/notification_service/cli/commands/server_command.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            'stream': 'ext://sys.stderr',
+            'formatter': 'default'
+        }
+    },
+    'root': {
+        'level': 'INFO',
+        'handlers': ['console']
+    }
+})
+
```
Comment (wuchaochen) [inline lib/notification_service/notification_service/cli/commands/server_command.py:45]: May need to change place.

File: lib/notification_service/notification_service/cli/cli_parser.py [hunk 3]
```diff
@@ -126,6 +126,13 @@ class GroupCommand(NamedTuple):
                                 lazy_load_command("notification_service.cli.commands.version_command.version"),
                                 [],
                                 "Shows the version of Notification.")
+
+SERVER_COMMAND = ActionCommand("server",
+                               "Start the notification server in foreground.",
```
Comment (SteNicholas) [inline lib/notification_service/notification_service/cli/cli_parser.py:131]: Removes the `.`.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: lib/notification_service/tests/cli/commands/test_server_command.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    @classmethod
+    def setUpClass(cls):
+        cls.parser = cli_parser.get_parser()
+
+    def test_cli_server_start(self):
+        os.environ["NOTIFICATION_HOME"] = os.path.join(os.path.dirname(__file__), "..", "..")
+        db_command.upgrade(self.parser.parse_args(['db', 'upgrade']))
+        server_process = multiprocessing.Process(target=server_command.server,
+                                                 args=(self.parser.parse_args(['server']),), )
+        server_process.start()
```
Comment (SteNicholas) [inline lib/notification_service/tests/cli/commands/test_server_command.py:41]: You also need to check the print of the command execution.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: lib/notification_service/notification_service/cli/commands/server_command.py (hunk 2 above)
Comment (Sxnan) [reply lib/notification_service/notification_service/cli/commands/server_command.py:31]: It is moved to `settings.py`

File: lib/notification_service/bin/start-notification.sh (hunk 1 above)
Comment (Sxnan) [reply lib/notification_service/bin/start-notification.sh]: notification server start added

File: lib/notification_service/notification_service/cli/commands/server_command.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    # We raise KeyboardInterrupt so that the server can stop gracefully
+    raise KeyboardInterrupt()
+
+
+def server(args):
+    """Start the notification server"""
+    signal.signal(signal.SIGTERM, sigterm_handler)
+    config_file_path = get_configuration_file_path()
+    server_runner = NotificationServerRunner(config_file_path)
+    server_runner.start(True)
```
[the remark this answers is not in the document]
Comment (Sxnan) [reply lib/notification_service/notification_service/cli/commands/server_command.py]: logging of pid is added in NotificationServerRunner

File: lib/notification_service/tests/cli/commands/test_server_command.py (hunk 4 above)
Comment (Sxnan) [reply lib/notification_service/tests/cli/commands/test_server_command.py]: Check is added

**Comment by SteNicholas** [comment]
@Sxnan , IMO, this pull request supports the damon start, could this provide stop subcommand?

File: lib/notification_service/notification_service/cli/cli_parser.py [hunk 6]
```diff
@@ -249,13 +254,24 @@ class GroupCommand(NamedTuple):
     )
 )
 
+SERVER_COMMANDS = (
+    ActionCommand("start",
+                  "Start the notification server",
```
Comment (SteNicholas) [inline lib/notification_service/notification_service/cli/cli_parser.py:259]:
```suggestion
                  "Starts the notification server",
```

File: lib/notification_service/notification_service/cli/cli_parser.py [hunk 7]
```diff
@@ -249,13 +254,24 @@ class GroupCommand(NamedTuple):
...
+                  lazy_load_command("notification_service.cli.commands.server_command.server_start"),
+                  [ARG_SERVER_DAEMON],
+                  "Start the notification server"),
```
Comment (SteNicholas) [inline lib/notification_service/notification_service/cli/cli_parser.py:262]:
```suggestion
                  "Starts the notification server"),
```

File: lib/notification_service/notification_service/server.py [hunk 8]
```diff
@@ -113,6 +115,8 @@ def _init_server(self):
                                              port=int(self.config.port))
 
     def start(self, is_block=False):
+        logger.info("Starting notification server with config: {}".format(self.config))
```
Comment (SteNicholas) [inline lib/notification_service/notification_service/server.py:118]: IMO, it's recommended to log the config file path and the detail of the config.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: lib/notification_service/notification_service/server.py (hunk 8 above)
Comment (Sxnan) [reply lib/notification_service/notification_service/server.py:118]: The `NotificationServerRunner` does not know where the log is. When running in the foreground, the log is printing to the stdout.

Comment (SteNicholas) [reply lib/notification_service/notification_service/server.py:118]: I mean the the path of the notification config, not the log.

Comment (Sxnan) [reply lib/notification_service/notification_service/server.py:119]: It has been added in cb190aa

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Pull request merged**